core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -252,11 +252,36 @@ public void testFailure() {
     assertThat(new File(newManifest.path())).doesNotExist();
   }
 
+  @TestTemplate
+  public void testIncreaseNumRetries() {
+    TestTables.TestTableOperations ops = table.ops();
+    ops.failCommits(TableProperties.COMMIT_NUM_RETRIES_DEFAULT + 1);
+
+    AppendFiles append = table.newFastAppend().appendFile(FILE_B);
+
+    // Default number of retries results in a failed commit
+    assertThatThrownBy(append::commit)
+        .isInstanceOf(CommitFailedException.class)
+        .hasMessage("Injected failure");
+
+    // After increasing the number of retries the commit succeeds
+    table
+        .updateProperties()
+        .set(
+            TableProperties.COMMIT_NUM_RETRIES,
+            String.valueOf(TableProperties.COMMIT_NUM_RETRIES_DEFAULT + 1))
+        .commit();
+
+    append.commit();
+
+    validateSnapshot(null, readMetadata().currentSnapshot(), FILE_B);
+  }
+
   @TestTemplate
   public void testAppendManifestCleanup() throws IOException {
-    // inject 5 failures
+    // inject failures
     TestTables.TestTableOperations ops = table.ops();
-    ops.failCommits(5);
+    ops.failCommits(TableProperties.COMMIT_NUM_RETRIES_DEFAULT + 1);
```

Comment:
TBH I don't see much value in updating this and the other tests

Reply:
The value here is that we can test boundary values for the config and not just use a value that is within the range.

Reply:
Makes sense. I reverted the changes in existing tests.
Thanks for taking a look!